src/parser/druid/feral/modules/azeritetraits/WildFleshrending.js

```diff
@@ -0,0 +1,215 @@
+    debug && console.log(`swipeBonus from items: ${this.swipeBonus}`);
+  }
+
+  on_byPlayer_cast(event) {
+    if (event.ability.guid !== SPELLS.SHRED.id &&
+        !SWIPE_SPELLS.includes(event.ability.guid)) {
+      return;
+    }
+    this.attackPower = event.attackPower;
+  }
+
+  on_byPlayer_damage(event) {
+    if (HIT_TYPES_TO_IGNORE.includes(event.hitType)) {
+      // Attacks that don't hit cannot benefit from the trait's bonus
+      return;
+    }
+
+    const isShred = event.ability.guid === SPELLS.SHRED.id;
+    if (!isShred && !SWIPE_SPELLS.includes(event.ability.guid)) {
+      // Only interested in shred and swipe
+      return;
+    }
+
+    if (isShred) {
+      this.shredsTotal += 1;
+    } else {
+      this.swipesTotal += 1;
+    }
+
+    const target = this.enemies.getEntities()[event.targetID];
+    if (!target || (
+        !target.hasBuff(SPELLS.THRASH_FERAL.id, event.timestamp, 0, 0, this.owner.playerId) &&
+        !target.hasBuff(SPELLS.THRASH_BEAR_DOT.id, event.timestamp, 0, 0, this.owner.playerId))) {
+      // Only applies bonus damage if hitting a target with player's Thrash (either cat or bear version) active on it
+      debug && console.log(`${this.owner.formatTimestamp(event.timestamp, 3)} hit without Thrash`);
+      return;
+    }
+
+    if (isShred) {
+      this.shredsWithThrash += 1;
+    } else {
+      this.swipesWithThrash += 1;
+    }
+
+    /**
+     * Through in-game testing I've found that the Wild Fleshrending trait's damage is applied like this:
+     * damageDone = (attackPower * ABILITY_COEFFICIENT + traitBonus) * modifier
+     * 
+     * The important thing is that there's just one modifier term applied to both the ability's base damage and to the trait's bonus damage. 
+     * The modifier includes things like shred's 20% buff against bleeding targets, Tiger's Fury, Savage Roar, the Feral Druid aura that's used to adjust DPS balance, and even the damage increase from a critical strike. It also includes things that are difficult to see from our logs such as Mystic Touch or encounter-specific effects like MOTHER taking extra damage in the 3rd chamber. The effect of enemy armor is also included within the modifier.
+     * We know the damageDone, attackPower, ABILITY_COEFFICIENT, and traitBonus which means we can calculate the modifier's value for this attack. With the modifier known we can multiply that by traitBonus to find how much damage it actually contributed to this attack.
+     * 
+     * Assumptions: There are no effects in play that adjust damage by a set number rather than by multiplying, apart from those that appear in the log as absorbed damage. Everything that modifies the ability's base damage also modifies the trait's bonus damage in the same way. I've yet to find anything that violates these assumptions.
+     */
+    
+    let coefficient = 1.0;
+    switch (event.ability.guid) {
```

Comment:
would this be better by attaching coefficients to the spellbook and then doing something like `this.abilities.getAbility(event.ability.guid).coefficient`?

Reply:
I don't know if this works a lot differently than the mage trait Whiteout, but we avoided having to use any spell coefficients by doing something similar, but slightly different logic:

https://github.com/WoWAnalyzer/WoWAnalyzer/blob/master/src/parser/mage/frost/modules/traits/Whiteout.js#L79

It makes one key assumption; the traits damage (from calculateAzeriteEffects) is only changed by four things: versatility, spec aura, critting, and external modifiers (like boss phases, chaos brand, etc).

Reply:
One other note on the PR - should there really be a suggestion for this trait? It feels like Thrash efficiency is a problem on its own, so shouldn't that be its own module (if it isnt one already)? I feel like you'd just have
"You didnt keep thrash up on the target"
"You lost fleshrending damage because you didn't keep thrash up on the target"

Which isn't really two suggestions

Reply:
**Coefficients**
Putting the coefficients into the spellbook instead of the spec's constants is a good idea. As you show, that'll allow them to be accessed more directly and it'll keep spell-specific information in one place. (DoT duration information is currently stored in Feral's `constants.js` which could benefit from being moved too.)

**Whiteout-style Calculation**
I think that method would work here too. Wild Fleshrending's damage is modified by quite a lot more personal effects than Whiteout's though, so it would be a more complex implementation. 

As an experiment I've implemented that "mage-style" approach too, and they mostly give the exact same result. I'll write more comparing them in the main discussion thread of this PR.

**Thrash Uptime Suggestion**
Wild Fleshrending does somewhat change the Feral rotation. Thrash is designed as an AoE ability and on single target the resources used to keep the bleed up could be used to do almost exactly the same damage directly (varies slightly depending on the player's secondary stats). In some conditions it's a damage loss to keep Thrash active as the direct damage alternatives become temporarily more efficient, so a simple "aim for full uptime" check wouldn't suit Thrash.

Even with Wild Fleshrending traits there can be times in the rotation where it's beneficial to let Thrash's DoT drop off then reapply it when you're ready to spam Shred/Swipe. So I think it's best to check for how well the Wild Fleshrending effect is being applied rather than looking directly at uptime.

Reply:
Meh, with so many modifiers (now that I look at yours, I'm pretty sure I missed one or two as well) it's hard to say if that's even easier to maintain than the coefficients. I'll let you and others decide which is better when it comes to actually maintaining it The real issue here is not having this data mined from the ability database. Hence the bounty on that... So much could be made simpler if we grab all of this data and update it automatically.